extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -485,9 +484,28 @@ export class ProjectsController {
 		return addDatabaseReferenceDialog;
 	}
 
-	public async addDatabaseReferenceCallback(project: Project, settings: ISystemDatabaseReferenceSettings | IDacpacReferenceSettings): Promise<void> {
+	public async addDatabaseReferenceCallback(project: Project, settings: ISystemDatabaseReferenceSettings | IDacpacReferenceSettings | IProjectReferenceSettings): Promise<void> {
 		try {
-			if ((<ISystemDatabaseReferenceSettings>settings).systemDb !== undefined) {
+			if ((<IProjectReferenceSettings>settings).projectName !== undefined) {
+				// get project path and guid
+				const projectReferenceSettings = settings as IProjectReferenceSettings;
+				const referencedProject = this.projects.find(p => p.projectFileName === projectReferenceSettings.projectName);
+				const relativePath = path.relative(project.projectFolderPath, referencedProject?.projectFilePath!);
+				projectReferenceSettings.projectRelativePath = vscode.Uri.file(relativePath);
+				projectReferenceSettings.projectGuid = referencedProject?.projectGuid!;
+
+				// check for cirular dependency
+				for (let r of referencedProject?.databaseReferences!) {
```

Comment:
a bit cleaner to use a filter clause here instead of nesting the `if` in the `for`.